### ISSUE TITLE
Cannot use Eigen matrices in header file.

### ISSUE THREAD

**Issue by aadhi0319** [nasa/trick#1204]
I am trying to use matrices from the Eigen C++ library in a TRICK model's header (rocket.hpp) file. The library is stored locally because I want the libraries to be included in the Github repository for the project. The file structure is as follows:

```
trick_sims
|__6DOF
     |__models/rocket/include/rocket.hpp
     |__models/rocket/src/rocketAnalytics.cpp
     |__models/...
     |__S_define
     |__S_overides.mk
|__dependencies
     |__Eigen
          |__Dense
```

The S_overrides.mk has the following variables:
TRICK_CFLAGS += -Imodels -I../dependencies -std=c++11
TRICK_CXXFLAGS += -Imodels -I../dependencies -std=c++11

In `rocket.hpp`, I want my rocket's state to be stored as an Eigen::matrix. To do this, I include the library in `rocket.hpp` with `#include <Eigen/Dense>`. This causes a slew of errors as TRICK attempts to process the entire Eigen library. What is interesting is that moving `#include <Eigen/Dense>` to any *.cpp file fixes all issues. Further, adding `-I/usr/local/include/eigen3` and removing `-I../dependencies` removes all issues even if `#include <Eigen/Dense>` is in `rocket.hpp`. This is odd because it is the exact same Eigen library in both directories (copied over with `cp -r /usr/local/include/eigen3/Eigen ../dependencies`). I have even tried experimenting with the TRICK_EXCLUDE, TRICK_ICG_EXCLUDE, and TRICK_SWIG_EXCLUDE parameter with no success. I am new to TRICK and I don't know what is wrong but I need to be able to access library defined datatypes (like `Eigen::matrix`) in header files. Thank you in advance for any help.

**Comment by jmpenn** [comment]
This may have been fixed by a recent fix that addresses issue #660.

**Comment by aadhi0319** [comment]
TRICK_EXCLUDE works to exclude Eigen now.
